src/libraries/Microsoft.PowerFx.Core/Functions/IAsyncTexlFunction4.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    // A Texl function capable of async invokes, using TimeZoneInfo and IRContext. 
+    internal interface IAsyncTexlFunction4
+    {
+        Task<FormulaValue> InvokeAsync(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args, CancellationToken cancellationToken);
```

Comment:
>TimeZoneInfo [](http://example.com/codeflow?start=39&length=12)

Are we going to need another function in the future that needs something other than the TimeZoneInfo (such as FormattingInfo)? If so, can we wrap those into something like "RuntimeSettings" so that we don't have to keep updating those interfaces?

Reply:
I know it's not ideal
FormattingInfo in only available in the Interpreter (and I don't like it)
I'd prefer the ServiceProvider approach like for connectors
Maybe I can avoid that interface in using the connector one... (even if it's not a connector)

Reply:
After looking more closely IAsyncConnectorTexlFunction is not allowing us to pass the IRContext.FormulaType
We'll need to reconcile all these signatures in future